### PR TITLE
Feat/invt cpc 741 search products by fields good pt1

### PR DESCRIPTION
**Pull request by emilegirardGit**
JIRA: https://champlainsaintlambert.atlassian.net/browse/CPC-741
## Context:
What is the ticket about and why are we doing this change.
## Changes
Add GetAllInventory - Backend
Add GetAllProductsInInventoryByFields - Back end
## Before and After UI (Required for UI-impacting PRs)
If this is a change to the UI, include before and after screenshots to show the differences.
If this is a new UI feature, include screenshots to show reviewers what it looks like. 
## Dev notes (Optional)
- Specific technical changes that should be noted
## Linked pull requests (Optional)
- pull request link

### CODE REVIEW

**Review by SandwichKing978 — DISMISSED** [review]

**Review by Eli293 — CHANGES_REQUESTED** [review]
Need to change inventoryType to enum instead of string to avoid merge conflicts

**Comment by nic5694** [comment]
Their seems to be an issue with the ./gradlew clean build try and run it to see if it can pin point to an error

File: api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/InventoryServiceClient.java [hunk 1]
```diff
@@ -52,18 +55,39 @@ public Mono<InventoryResponseDTO> addInventory(final InventoryRequestDTO model){
...
+                .accept(MediaType.APPLICATION_JSON)
+                .retrieve()
+                .bodyToFlux(ProductResponseDTO.class);
+    }
+
+    public Flux<InventoryResponseDTO> getAllInventory(){
+        return webClient.get()
+                .uri(inventoryServiceUrl)
+                .accept(MediaType.APPLICATION_JSON)
+                .retrieve()
```
Comment (nic5694) [inline api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/InventoryServiceClient.java:88]: You should maybe specify the return, and check the number of items in the list to ensure correct and profound testing. Especially with the response EventTextStream to make sure its returning a FLUX

File: api-gateway/src/test/java/com/petclinic/bffapigateway/domainclientlayer/InventoryServiceClientIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    @AfterAll
+    static void tearDown() throws IOException{
+        mockWebServer.shutdown();
+    }
+
+    @Test
+    void getProductsInInventoryByInventoryIdAndProductsField() throws JsonProcessingException {
+        ProductResponseDTO productResponseDTO = new ProductResponseDTO(
```
Comment (nic5694) [inline api-gateway/src/test/java/com/petclinic/bffapigateway/domainclientlayer/InventoryServiceClientIntegrationTest.java:50]: maybe use the builder?

File: inventory-service/src/test/java/com/petclinic/inventoryservice/businesslayer/ProductInventoryServiceUnitTest.java [hunk 3]
```diff
@@ -46,14 +45,171 @@ class ProductInventoryServiceUnitTest {
...
+                        null,
+                        null,
+                        null);
+
+        StepVerifier
+                .create(productResponseDTOFlux)
+                .expectNextCount(2)
+                .verifyComplete();
+    }*/
+
```
Comment (nic5694) [inline inventory-service/src/test/java/com/petclinic/inventoryservice/businesslayer/ProductInventoryServiceUnitTest.java:87]: Remove any tests that are not going to be used, its best to not push commented out code to main

File: inventory-service/src/test/java/com/petclinic/inventoryservice/presentationlayer/InventoryControllerIntegrationTest.java [hunk 4]
```diff
@@ -85,6 +106,236 @@ void addProductToInventory_WithInvalidInventoryIdAndValidBody_ShouldThrowNotFoun
...
+
+
+        webTestClient.get()
+                .uri("/inventory/{inventoryId}/products?productPrice={productPrice}",
+                        invalidInventoryId, invalidProductPrice)
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectStatus().isNotFound()
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBody()
```
Comment (nic5694) [inline inventory-service/src/test/java/com/petclinic/inventoryservice/presentationlayer/InventoryControllerIntegrationTest.java:246]: check that it throws that not found exception class maybe?

**Review by nic5694 — CHANGES_REQUESTED** [review]
check the comments i left and check the build error. Consider removing commented out code that is not useful its best to not push commented code to our main branch.

**Review by nic5694 — DISMISSED** [review]

**Review by Eli293 — APPROVED** [review]
Looks good to me.

**Review by SandwichKing978 — APPROVED** [review]
Looks good

**Pull request merged**